demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1384 @@
+ * If false, tries to establish the existing session if there was session
+ * already present in broker.
+ * @param[out] pSessionPresent Session was already present in the broker or not.
+ * Session present response is obtained from the CONNACK from broker.
+ *
+ * @return EXIT_SUCCESS if an MQTT session is established;
+ * EXIT_FAILURE otherwise.
+ */
+static int establishMqttSession( MQTTContext_t * pMqttContext );
+
+/**
+ * @brief Close an MQTT session by sending MQTT DISCONNECT.
+ *
+ * @param[in] pMqttContext MQTT context pointer.
+ *
+ * @return EXIT_SUCCESS if DISCONNECT was successfully sent;
+ * EXIT_FAILURE otherwise.
+ */
+static int disconnectMqttSession( MQTTContext_t * pMqttContext );
+
+/**
+ * @brief Publish message to a topic.
+ *
+ * This function publishes a message to a given topic & QoS.
+ *
+ * @param[in] pacTopic Mqtt topic filter.
+ *
+ * @param[in] topicLen Length of the topic filter.
+ *
+ * @param[in] pMsg Message to publish.
+ *
+ * @param[in] msgSize Message size.
+ *
+ * @param[in] qos Quality of Service
+ *
+ * @return OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttPublish( const char * const pacTopic,
+                                    uint16_t topicLen,
+                                    const char * pMsg,
+                                    uint32_t msgSize,
+                                    uint8_t qos );
+
+/**
+ * @brief Subscribe to the MQTT topic filter, and registers the handler for the topic filter with the subscription manager.
+ *
+ * This function subscribes to the Mqtt topics with the Quality of service
+ * received as parameter. This function also registers a callback for the
+ * topicfilter.
+ *
+ * @param[in] pTopicFilter Mqtt topic filter.
+ *
+ * @param[in] topicFilterLength Length of the topic filter.
+ *
+ * @param[in] qos Quality of Service
+ *
+ * @param[in] callback Callback to be registered for handling an incoming PUBLISH message on the topic.
+ *
+ * @return OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttSubscribe( const char * pTopicFilter,
+                                      uint16_t topicFilterLength,
+                                      uint8_t qos,
+                                      OtaMqttCallback_t callback );
+
+/**
+ * @brief Unsubscribe to the Mqtt topics.
+ *
+ * This function unsubscribes to the Mqtt topics with the Quality of service
+ * received as parameter.
+ *
+ * @param[in] pTopicFilter Mqtt topic filter.
+ *
+ * @param[in] topicFilterLength Length of the topic filter.
+ *
+ * @param[qos] qos Quality of Service
+ *
+ * @return  OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttUnsubscribe( const char * pTopicFilter,
+                                        uint16_t topicFilterLength,
+                                        uint8_t qos );
+
+/**
+ * @brief Start OTA demo.
```

Comment:
More information about the operations in this function will be valuable for the reader

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/7bed0e4558dfd4a6c985fc6da34260501e76d4c7